crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java

```diff
@@ -80,8 +81,9 @@ private static OutputStream createApacheEncryptedStream(OutputStream output, Key
         return new CtrCryptoOutputStream(PROPS, output, secretKey.getEncoded(), iv);
     }
 
-    private static OutputStream createDefaultEncryptedStream(OutputStream output, String algorithm) {
-        SeekableCipher cipher = SeekableCipherFactory.getCipher(algorithm);
+    private static OutputStream createDefaultEncryptedStream(OutputStream output, KeyMaterial keyMaterial,
+            String algorithm) {
+        SeekableCipher cipher = SeekableCipherFactory.getCipher(algorithm, keyMaterial);
```

Comment:
the SeekableCipherFactory is dangerous. would either remove the method that does generates key-material, or rename it to getCipherWithGeneratedKeyMaterial, or make the key material an `Optional<>` parameter.